### PR TITLE
388 delete user endpoint

### PR DESCRIPTION
**Pull request by ganznz**

### CODE REVIEW

File: server/src/middleware/routes.test.ts [hunk 1]
```diff
@@ -334,6 +334,57 @@ describe("Endpoints", () => {
...
+        .set("Authorization", `Bearer ${adminToken}`)
+        .send({ uid: MEMBER_USER_UID })
+
+      expect(res.status).toEqual(200)
+      const deletedUser = await new UserDataService().getUserData(
+        MEMBER_USER_UID
+      )
+      expect(deletedUser).toEqual(undefined)
+      const unaffectedUser = await new UserDataService().getUserData(
+        GUEST_USER_UID
```
Comment (bcho892) [inline server/src/middleware/routes.test.ts:361]: we should check if the user is actually deleted in auth, as that will cause problems if they do

note the uid is same for both auth and firestore

Comment (bcho892) [reply server/src/middleware/routes.test.ts:361]: idk if the `AuthService` already has a method for that but if not can just use the `firebase-admin` package in the tests

File: server/src/service-layer/controllers/UserController.ts [hunk 2]
```diff
@@ -56,4 +60,31 @@ export class UsersController extends Controller {
...
+        const authService = new AuthService()
+        const userDataService = new UserDataService()
+
+        const userClaims = await authService.getCustomerUserClaim(userUid)
+        if (userClaims.admin) {
+          this.setStatus(403) // forbidden request
+          return { error: "Cannot delete another admin." }
+        }
+
+        await authService.deleteUser(userUid)
```
Comment (bcho892) [inline server/src/service-layer/controllers/UserController.ts:82]: missing success code

**Comment by ganznz** [comment]
im sorry benson idk what ive done and now codegen check is failing 🥺🥺🥺

File: server/src/middleware/__generated__/swagger.json [hunk 3]
```diff
@@ -553,7 +577,7 @@
 					},
 					"nextCursor": {
 						"type": "string",
-						"description": "Needed for firestore operations which do not support offset\nbased pagination\n\n**Will be undefined in case of last page**"
+						"description": "Needed for firestore operations which do not support offset\r\nbased pagination\r\n\r\n**Will be undefined in case of last page**"
```
Comment (bcho892) [inline server/src/middleware/__generated__/swagger.json:580]: Bruh it's doing this again, I'll try investigate why it does this but for now just need to remove all the `\r` (it's a Linux vs windows moment)

**Review by bcho892 — APPROVED** [review]
LGTM!

**Pull request merged**